plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -191,18 +192,29 @@ func (p *plugin) replaceValuesInline() error {
 	if err != nil {
 		return err
 	}
-	chValues := make(map[string]interface{})
-	if err = yaml.Unmarshal(pValues, &chValues); err != nil {
+	chValues, err := kyaml.Parse(string(pValues))
+	if err != nil {
 		return err
 	}
+	inlineValues, err := kyaml.FromMap(p.ValuesInline)
+	if err != nil {
+		return err
+	}
+	var outValues *kyaml.RNode
 	switch p.ValuesMerge {
 	case valuesMergeOptionOverride:
-		err = mergo.Merge(
-			&chValues, p.ValuesInline, mergo.WithOverride)
+		outValues, err = merge2.Merge(inlineValues, chValues, kyaml.MergeOptions{})
 	case valuesMergeOptionMerge:
-		err = mergo.Merge(&chValues, p.ValuesInline)
+		outValues, err = merge2.Merge(chValues, inlineValues, kyaml.MergeOptions{})
+	}
```

Comment:
Good point! I added the explanation in the code.

Unrelated, but I also decided to copy the value before merging, so that the original values are preserved after the merge. Someone might still want to use it.